tests/integration/test_config.py

```diff
@@ -0,0 +1,101 @@
+        charm = await ops_test.build_charm(".")
+        await ops_test.model.deploy(
+            charm,
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+        )
+        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=1500)
+
+    leader_unit = await get_leader_unit(ops_test, DATABASE_APP_NAME)
+    test_string = "abcXYZ123"
+
+    configs = [
+        {
+            "durability_synchronous_commit": {test_string, "on"}
+        },  # config option is one of `on`, `remote_apply` or `remote_write`
+        {
+            "instance_password_encryption": {test_string, "md5"}
+        },  # config option is one of `md5` or `scram-sha-256`
+        {
+            "logging_log_min_duration_statement": {"2147483648", "1"}
+        },  # config option is between -1 and 2147483647
+        {
+            "memory_maintenance_work_mem": {"2147483648", "1024"}
+        },  # config option is between 1024 and 2147483647
+        {"memory_max_prepared_transactions": {"-1", "0"}},  # config option is between 0 and 262143
+        {"memory_shared_buffers": {"15", "16"}},  # config option is greater or equal than 16
+        {"memory_temp_buffers": {"99", "100"}},  # config option is between 100 and 1073741823
+        {"memory_work_mem": {"2147483648", "64"}},  # config option is between 64 and 2147483647
+        {
+            "optimizer_constraint_exclusion": {test_string, "off"}
+        },  # config option is one of `on`, `off` or `partition`
+        {
+            "optimizer_default_statistics_target": {"0", "1"}
+        },  # config option is between 1 and 10000
+        {"optimizer_from_collapse_limit": {"0", "1"}},  # config option is between 1 and 2147483647
+        {"optimizer_join_collapse_limit": {"0", "1"}},  # config option is between 1 and 2147483647
+        {"profile": {test_string, "testing"}},  # config option is one of `testing` or `production`
+        {"profile_limit_memory": {"127", "128"}},  # config option is between 128 and 9999999
+        {
+            "response_bytea_output": {test_string, "hex"}
+        },  # config option is one of `escape` or `hex`
+        {
+            "vacuum_autovacuum_analyze_scale_factor": {"-1", "0"}
+        },  # config option is between 0 and 100
+        {
+            "vacuum_autovacuum_vacuum_scale_factor": {"-1", "0"}
+        },  # config option is between 0 and 100
+        {
+            "vacuum_autovacuum_analyze_threshold": {"-1", "0"}
+        },  # config option is between 0 and 2147483647
+        {
+            "vacuum_autovacuum_freeze_max_age": {"99999", "100000"}
+        },  # config option is between 100000 and 2000000000
+        {
+            "vacuum_autovacuum_vacuum_cost_delay": {"-2", "-1"}
+        },  # config option is between -1 and 100
+        {
+            "vacuum_vacuum_freeze_table_age": {"-1", "0"}
+        },  # config option is between 0 and 2000000000
```

Comment:
Sets are not ordered, so when you create a list from them there is no guarantee that the order will be kept. Best to just use lists from the get go.